### PR TITLE
Support for CloudFormation Conditions in Convection

### PR DESCRIPTION
**Pull request by simonirwin-r7**
This pull request adds native Convection support for CF Conditions.  

README.md updated accordingly.

Tests added:
1) condition blocks generate the correct JSON
2) test that resources can use conditions
3) test that resource properties can use conditions

### CODE REVIEW

File: convection.gemspec [hunk 1]
```diff
@@ -22,6 +22,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rake', '~> 10.0'
   spec.add_development_dependency 'thor-scmversion', '= 1.7.0'
   spec.add_development_dependency 'minitest'
+  spec.add_development_dependency 'simplecov'
```
Comment (athompson-r7) [inline convection.gemspec]: Thanks for adding this!  I'd expect our coverage to be much lower than it is.

**Comment by athompson-r7** [comment]
:+1: @jmanero-r7 do you want to take a look aswell?

**Comment by simonirwin-r7** [comment]
Coverage was recorded as ~59% before I had it setup correctly. It was counting everything except the actual executable parts (!) of all the methods. Once setup, it moved to ~65%. I think it's being generous 😊. Tbh I thought the coverage committee was a separate pull request which I haven't made it yet - I only merged it to my forked master branch before leaving the office.

**Comment by jmanero-r7** [comment]
:+1: That's awesome!

**Comment by jmanero-r7** [comment]
@simonirwin-r7 Do you have permission to merge this yet?

**Comment by simonirwin-r7** [comment]
It's showing me a 'Merge branch' button, so I guess I can merge and Copernicus ok with you guys?

**Comment by simonirwin-r7** [comment]
'close branch' becomes Copernicus in auto-correct !

**Comment by simonirwin-r7** [comment]
'close pull request', I mean...think I should put down the phone. Sorry for confusion.

**Pull request merged**